nifi-asymmetrik-standard-bundle/nifi-asymmetrik-standard-processors/src/main/java/com/asymmetrik/nifi/processors/aws/ListS3Batch.java

```diff
@@ -0,0 +1,450 @@
+            S3_BUCKET,
+            S3_PREFIX,
+            S3_START_PREFIX,
+            S3_END_PREFIX,
+            BATCH_SIZE,
+            FETCH_OWNER,
+            OVERRIDE_ENDPOINT,
+            SSL_CONTEXT_SERVICE
+        );
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        this.client = getClient(context);
+    }
+
+    @OnStopped
+    public void onStopped() {
+        if (this.client != null) {
+            this.client.shutdown();
+            this.client = null;
+        }
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        final String region = context.getProperty(AWS_REGION).getValue();
+        final boolean fetchOwner = context.getProperty(FETCH_OWNER).asBoolean();
+        final String bucketName = context.getProperty(S3_BUCKET)
+                .evaluateAttributeExpressions()
+                .getValue();
+        final String prefix = context.getProperty(S3_PREFIX)
+                .evaluateAttributeExpressions()
+                .getValue();
+        final String startKey = context.getProperty(S3_START_PREFIX)
+                .evaluateAttributeExpressions()
+                .getValue();
+        final String endKey = context.getProperty(S3_END_PREFIX)
+                .evaluateAttributeExpressions()
+                .getValue();
+        final int batchSize = context.getProperty(BATCH_SIZE)
+                .evaluateAttributeExpressions()
+                .asInteger();
+
+        final String lastConfig = getItemFromContext(context, LAST_CONFIG);
+        final String config = region + "," + bucketName + "," + prefix + "," + startKey + "," + endKey + "," + fetchOwner;
+        final boolean propsHaveNotChanged = config.equals(lastConfig);
+
+        // If the configuration properties have not changed, Yield if the current listing is complete
+        if (propsHaveNotChanged && isComplete(context)) {
+            context.yield();
+            return;
+        }
+        
+        // If the configuration properties have not changed, find where we left off in any prior iteration
+        // Otherwise set the starting list point by the configured prefix and starting key
+        final String startAfter = propsHaveNotChanged
+                ? getItemFromContext(context, LAST_ELEMENT_KEY)
+                : prefix + startKey;
+        
+        // Create the List S3 request
+        final ListObjectsV2Request request = new ListObjectsV2Request()
+                .withBucketName(bucketName)
+                .withPrefix(prefix)
+                .withStartAfter(startAfter)
+                .withMaxKeys(batchSize)
+                .withFetchOwner(fetchOwner);
+
+        // Execute the listS3 request
+        final ListObjectsV2Result listResponse = client.listObjectsV2(request);
+        List<S3ObjectSummary> s3obj = listResponse.getObjectSummaries();
+        
+        if (s3obj.isEmpty()) {
```

Comment:
Does getObjectSummaries always return an empty list, or will it return null?

Reply:
Added null check, just in case the AWS API ever messes up.